crates/web5/src/credentials/status_list_credential.rs

```diff
@@ -0,0 +1,560 @@
+            None => Vec::new(),
+        };
+
+        // Generate the base64 bitstring from the status list indexes.
+        let base64_bitstring = Self::bitstring_generation(status_list_indexes)?;
+
+        // Construct the properties for the credential subject.
+        let additional_properties = {
+            let mut properties = HashMap::new();
+            properties.insert(
+                "statusPurpose".to_string(),
+                JsonValue::String(status_purpose.clone()),
+            );
+            properties.insert(
+                "type".to_string(),
+                JsonValue::String(STATUS_LIST_2021.to_string()),
+            );
+            properties.insert(
+                "encodedList".to_string(),
+                JsonValue::String(base64_bitstring.clone()),
+            );
+            JsonObject { properties }
+        };
+
+        let credential_subject = CredentialSubject {
+            id: format!("urn:uuid:{}", uuid::Uuid::new_v4()),
+            additional_properties: Some(additional_properties),
+        };
+
+        let vc_options = VerifiableCredentialCreateOptions {
+            id: Some(format!("urn:uuid:{}", uuid::Uuid::new_v4())),
+            context: Some(vec![STATUS_LIST_CREDENTIAL_CONTEXT.to_string()]),
+            r#type: Some(vec![STATUS_LIST_CREDENTIAL_TYPE.to_string()]),
+            issuance_date: None,
+            expiration_date: None,
+            credential_status: None,
+            credential_schema: None,
+            evidence: None,
+        };
+
+        let verifiable_credential =
+            VerifiableCredential::create(issuer, credential_subject, Some(vc_options))?;
+
+        Ok(Self {
+            base: verifiable_credential,
+        })
+    }
+
+    /// Checks if a given credential is disabled according to this Status List Credential.
+    ///
+    /// # Arguments
+    ///
+    /// * `credential` - The `VerifiableCredential` to check.
+    ///
+    /// # Returns
+    ///
+    /// * `Ok(true)` if the credential is disabled, `Ok(false)` otherwise.
+    /// * `Err` if the credential status is invalid or incompatible.
+    ///
+    /// # Example
+    ///
+    ///  let is_disabled = status_list_credential.is_disabled(&credential_to_check)?;
+    ///  println!("Credential is disabled: {}", is_disabled);
+    ///
+    pub fn is_disabled(&self, credential: &VerifiableCredential) -> Result<bool> {
+        let status = credential.credential_status.as_ref().ok_or_else(|| {
+            Web5Error::Parameter("no credential status found in credential".to_string())
+        })?;
+
+        // Check if the status type matches
+        if status.r#type != STATUS_LIST_2021_ENTRY {
+            return Err(Web5Error::Parameter(format!(
+                "unsupported status type: {}",
+                status.r#type
+            )));
+        }
+
+        // Check if the status purpose matches
+        let status_purpose = self
```

Comment:
for both `status_purpose` and `encoded_list` we've got these similar, super vertical blocks of code that do "get this string attribute from additional_properties". I think it'd be more readable and reusable to put logic in a private function that doesn't use as many `and_then`s.

Reply:
yup you are right, let me do this